### PR TITLE
Add the route53 libdns provider

### PR DESCRIPTION
**Pull request by danlsgiga**
This PR introduces the Route53 libdns provider

A few notes about the expected behaviours of the implementation:

- Route53 changes will return only after AWS validates the changes are `INSYNC`
- Since we are using the AWS Go SDK, all the authentication methods (Env vars, EC2 Instance Profile and Shared credential file) are supported out of the box
- Any application willing to use this library will need to initialize the AWS client session via `NewSession()`

WIP #1

### CODE REVIEW

File: route53/client.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	}
+
+	for _, rrset := range recordSets {
+		for _, rrsetRecord := range rrset.ResourceRecords {
+			record := libdns.Record{
+				Name:  *rrset.Name,
+				Value: *rrsetRecord.Value,
+				Type:  *rrset.Type,
+				TTL:   time.Duration(*rrset.TTL) * time.Second,
+			}
```
Comment (mholt) [inline route53/client.go:76]: Is there any sort of "ID" associated with a record, from the Route53-side? i.e. does Route53 assign its own IDs to records? If so, it should be set in this struct too.

File: route53/client.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		}
+	}
+
+	return "", fmt.Errorf("%s: No zones found for the domain %s", r53.ErrCodeHostedZoneNotFound, zoneName)
+}
+
+func (p *Provider) createRecord(ctx context.Context, zoneID string, record libdns.Record) (libdns.Record, error) {
+	// AWS Route53 TXT record value must be enclosed in quotation marks on create
+	if record.Type == "TXT" {
+		record.Value = strconv.Quote(record.Value)
```
Comment (mholt) [inline route53/client.go:119]: I did not know about this function -- very cool!

... dang, I wish I knew about this earlier.

File: route53/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+[![godoc reference](https://img.shields.io/badge/godoc-reference-blue.svg)](https://pkg.go.dev/github.com/libdns/route53)
+
+This package implements the [libdns interfaces](https://github.com/libdns/libdns) for AWS [Route53](https://aws.amazon.com/route53/).
+
+## Authenticating
+
+This package supports all the credential configuration methods described in the [AWS Developer Guide](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html), such as `Environment Variables`, `EC2 Instance Profile` and the `AWS Credentials file` located in `.aws/credentials`
+
+The following IAM policy is a minimal working example to give Caddy permissions to manage DNS records:
```
Comment (mholt) [inline route53/README.md:12]: One cool thing is that these libdns packages are totally unrelated to Caddy, so we don't need to make any references to Caddy; they can also be used by other code bases.

File: route53/client.go (hunk 1 above)
Comment (danlsgiga) [reply route53/client.go:76]: Yeah, I looked at that and they don't have any IDs for recordSets as per https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#ResourceRecordSet

The way it works is you need to submit a [changeBatch](https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#ChangeBatch) and in this struct you define all the the recordSet info... so for example, to DELETE a record, you submit a `changeBatch` with the `Name`, `Value`, `TTL` and `Type` of the recordSet and they do their sourcery under the hood.

File: route53/README.md (hunk 3 above)
Comment (danlsgiga) [reply route53/README.md:12]: Good point! Going to remove that reference

File: route53/client.go (hunk 1 above)
Comment (mholt) [reply route53/client.go:76]: Ah, interesting -- no problem then.

**Comment by danlsgiga** [comment]
hmm... not sure... I'm getting access denied when I try to push there

**Comment by mholt** [comment]
@danlsgiga It says the invite is still pending... need me to resend it?

**Comment by danlsgiga** [comment]
nvm... I probably need more coffee... forgot to check my email! 😄 Thanks
